ios: fix builds by using --openssl-legacy-provider

jakubgs opened the pull request:
Fixes issue with following failure:
```
Failed to construct transformer:  Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:130:10)
    at stableHash (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro-cache/src/stableHash.js:19:8)
    at JsTransformer.getCacheKey (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/JSTransformer/worker.js:471:7)
    at getTransformCacheKey (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/DeltaBundler/Transformer/getTransformCacheKey.js:39:29)
    at new Transformer (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/DeltaBundler/Transformer.js:147:28)
    at /Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/Bundler.js:54:29
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
```
https://roytuts.com/how-to-fix-err_ossl_evp_unsupported-in-react-js-application/

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | fff501a3 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-15167/1/) | 2023-02-22 16:46:05 | ~3 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-15167/1/consoleText) |

jakubgs commented:
It appears the issue is specific to `macm1-01-ms`, which is our last remaining MacStadium M1 host which I want to decommission.

For now I think it would be best to just take it offline.

jakubgs commented:
And the OpenSSL version is the same everywhere:
```
 > a ci-slave-macos -o -a 'source .profile && openssl version'      
maci7-01.ms-eu-dublin.ci.devel | CHANGED | rc=0 | (stdout) LibreSSL 3.3.6
macm1-02.he-eu-fsn1.ci.devel | CHANGED | rc=0 | (stdout) LibreSSL 3.3.6
maci7-02.ms-eu-dublin.ci.devel | CHANGED | rc=0 | (stdout) LibreSSL 3.3.6
macm1-01.he-eu-fsn1.ci.devel | CHANGED | rc=0 | (stdout) LibreSSL 3.3.6
maci7-03.ms-eu-dublin.ci.devel | CHANGED | rc=0 | (stdout) LibreSSL 3.3.6
macm1-01.ms-eu-dublin.ci.devel | CHANGED | rc=0 | (stdout) LibreSSL 3.3.6
```
The new one, but it shouldn't matter.

jakubgs committed:
ios: use --openssl-legacy-provider with iOS builds

Fixes issue with following failure:
```
Failed to construct transformer:  Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:130:10)
    at stableHash (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro-cache/src/stableHash.js:19:8)
    at JsTransformer.getCacheKey (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/JSTransformer/worker.js:471:7)
    at getTransformCacheKey (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/DeltaBundler/Transformer/getTransformCacheKey.js:39:29)
    at new Transformer (/Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/DeltaBundler/Transformer.js:147:28)
    at /Users/jenkins/workspace/status-mobile/platforms/ios/node_modules/metro/src/Bundler.js:54:29
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
```
https://roytuts.com/how-to-fix-err_ossl_evp_unsupported-in-react-js-application/

Signed-off-by: Jakub Sokołowski <jakub@status.im>

jakubgs commented:
Looks like adding `--openssl-legacy-provider ` flag in `NODE_ARGS` for `react-native` xcode script helps:
https://github.com/status-im/status-mobile/blob/fa1d396bf75ec656d2d315bed99e5d04d33926f3/ios/StatusIm.xcodeproj/project.pbxproj#L529

But this is not a permanent change as that line is managed by RN.
